src/osd/OSD.cc

```diff
@@ -3671,7 +3671,6 @@ void OSD::recursive_remove_collection(CephContext* cct,
   ghobject_t next;
   int max = cct->_conf->osd_target_transaction_size;
   vector<ghobject_t> objects;
-  objects.reserve(max);
```

Comment:
take `BlueStore::collection_list()` as an example, by moving `objects.reserve(max)` into `BlueStore::_collection_list()`,  `objects.reserve(max)` will be called multiple times here. so what's the merit of this change?